gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsHttpMonitor.java

```diff
@@ -148,23 +147,18 @@ private String queryCluster(ProxyBackendConfiguration backend, String path)
         Call call = client.newCall(request);
 
         try (Response res = call.execute()) {
-            switch (fromStatusCode(res.code())) {
-                case HttpStatus.OK:
-                    return res.body().string();
-                case HttpStatus.UNAUTHORIZED:
+            return switch (fromStatusCode(res.code())) {
+                case HttpStatus.OK -> requireNonNull(res.body(), "body is null").string();
+                case HttpStatus.UNAUTHORIZED -> {
                     log.info("Unauthorized to fetch cluster stats");
-                    log.debug("username: %s, targetUrl: %s, cookieStore: %s",
-                            username,
-                            targetUrl,
-                            client.cookieJar().loadForRequest(HttpUrl.parse(targetUrl)));
-                    return null;
-                default:
-                    return null;
-            }
+                    log.debug("username: %s, targetUrl: %s, cookieStore: %s", username, targetUrl, client.cookieJar().loadForRequest(HttpUrl.parse(targetUrl)));
+                    yield null;
+                }
+                default -> null;
+            };
         }
         catch (IOException e) {
-            e.printStackTrace();
+            throw new UncheckedIOException(e);
```

Comment:
I'm a bit confused on the error handling.
Why mix null and throw exception for error handling?